create_history_version.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+set -e # Set to stop execution on error
+
+charts_dir="charts"
+fluid_dir="$charts_dir/fluid"
+fluid_source_dir="../fluid/$charts_dir/fluid"
+history_dir="history-versions"
+
+# Ensure the current directory is the 'charts' directory
+if [ "$(basename $(pwd))" != "$charts_dir" ]; then
+  echo "Error: The script should be executed in the 'charts' directory."
+  exit 1
+fi
+
+# Ensure the current directory contains the 'charts' subdirectory
+if [ ! -d "$charts_dir" ]; then
+  echo "Error: The 'charts' directory does not exist in the current directory."
+  exit 1
+fi
+
+# Get the git branch of current working directory
+branch=$(git symbolic-ref --short HEAD)
+
+# Extract the version number from branch name
+version=$(echo $branch | sed 's/.*-\([0-9]\.[0-9]\.[0-9].*\)/\1/')
+
+# Check if there is no version number in branch name
+if [ -z "$version" ]; then
+  echo "Error: No version number found in branch name."
+  exit 1
+fi
+
+# Check whether the version number is valid
+if ! [[ "$version" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9_\.]+)*$ ]]; then
```

Comment:
Is that '*' should be '?' for the meaning of zero or once?

```suggestion
if ! [[ "$version" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9_\.]+)?$ ]]; then
```